src/_macosx.m

```diff
@@ -6208,6 +6207,33 @@ static void timer_callback(CFRunLoopTimerRef timer, void* info)
     Timer_new,                 /* tp_new */
 };
 
+static bool verify_framework(void)
+{
+#ifdef COMPILING_FOR_10_6
+    NSRunningApplication* app = [NSRunningApplication currentApplication];
+    NSApplicationActivationPolicy activationPolicy = [app activationPolicy];
+    switch (activationPolicy) {
+        case NSApplicationActivationPolicyRegular:
+        case NSApplicationActivationPolicyAccessory:
+            return true;
+        case NSApplicationActivationPolicyProhibited:
+            break;
```

Comment:
Could we add a default case handling -1? I.e. when this is called without an NSApplication? Perhaps raising a warning that the check will not work? Alternatively I think that we should at least document that this will not work without an NSApplication